pkg/clusters/addons.go

```diff
@@ -33,6 +33,10 @@ type Addon interface {
 	// Delete removes the addon component from the given cluster.
 	Delete(ctx context.Context, cluster Cluster) error
 
+	// DumpDiagnostics gathers and returns diagnostic information for an addon. Its return map is a map of string
+	// filenames to file content byte slices.
+	DumpDiagnostics(ctx context.Context, cluster Cluster) (map[string][]byte, error)
```

Comment:
Hopefully not to the point of overflowing memory? Offhand I can't think of anything we have now that would potentially need to dump a large binary resource, and text >4GB even seems unlikely--I don't think even our largest configs are >100MB.

It doesn't seem like this would buy us too much since [GitHub runners have neither a lot of RAM not a lot of disk](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources)--if we do end up with a file >7GB, we better hope we only have one.

f801487 will do it, but I'd prefer to revert it in light of the limitations above and have a single consistent means of returning data to the Cleaner.

Reply:
Not sure how I feel about the f801487 commit, I really was hoping we could provide some kind of interface which would hide the storage details and allow CRUD ops :thinking: 

For now though, and in the interest of time let's do what you suggest and revert it and just go with what you had originally: I think that getting something simple out that can actually produce results where before we had none is a great place to start.

Reply:
I think GitHub will keep the commit I obliterated in limbo somewhere, but if not, [addon_file_diag.diff.txt](https://github.com/Kong/kubernetes-testing-framework/files/9193461/addon_file_diag.diff.txt)